Move the NAT Gateways to the public subnets

jsf9k opened the pull request:
## 🗣 Description

I had previously put the NAT Gateways in the private subnets.  This is foolish, since then they cannot reach out to the internet!  This pull request fixes my bone-headed mistake.

## 💭 Motivation and Context

The private subnets cannot reach out to the internet without this fix, so it is very necessary.

## 🧪 Testing

I deployed these changes to production and verified that the private subnets can now reach out to the internet.

## 🚥 Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

jsf9k committed:
Move the NAT GWs to the public subnets

jsf9k committed:
Update the descriptions for the public/private subnet CIDR block variables

dav3r commented on subnets.tf:
```diff
@@ -43,6 +43,6 @@ resource "aws_nat_gateway" "nat_gws" {
 
   allocation_id = aws_eip.nat_gw_eips[each.value].id
-  subnet_id     = module.private.subnets[each.value].id
+  subnet_id     = module.public.subnets[var.public_subnet_cidr_blocks[index(var.private_subnet_cidr_blocks, each.value)]].id
```
This line is now a bit of a brain-twister, at least for me- it took me a minute to grep what was going on.  Can you please add a comment here to assist future readers?

jsf9k replied:
This was something I intended to do but forgot about.  See commit 3d80193.

dav3r reviewed (approved):
Approved, but I requested one comment be added.  🏞

jsf9k committed:
Add a comment explaining some Terraform code

This is the Terraform code that marries up the NAT GW for a private
subnet with a public subnet in the same AZ.

jsf9k committed:
Create an empty commit to make GitHub Actions run again

I am doing this after GitHub has recovered from an outage.

mcdonnnj reviewed (approved):
Nice work 👍
